agent/util/expansions.go

```diff
@@ -91,6 +91,15 @@ func (e *DynamicExpansions) PutAndRedact(key, value string) {
 	}
 }
 
+// Redact is used when the key and value shouldn't be an expansion,
+// but should be redacted like one.
+func (e *DynamicExpansions) Redact(key, value string) {
```

Comment:
I don't have a great suggestion here nor do I think it needs to be changed, just that this seems a little awkward since it has nothing to do with expansions at all, but is rather here just because expansions happen to have redaction functionality.

Might be sufficient to add a comment to the `DynamicExpansions` struct saying that, separately from the expansion functionality, there's also support for redacting strings that are sensitive but aren't actually expansions.